### ISSUE TITLE
Implement bookmark garbage collection

### ISSUE THREAD

**Issue by 0x-r4bbit** [status-im/status-go#2705]
As discussed in #2704, "deletion" of browser bookmarks merely means setting their `removed` flag to `true`, resulting in them not being actually deleted.

This will create memory pressure that needs to be accounted for. To solve this, we'll implement a simple garbage collection mechanism that is run every time the messenger bootstraps. The algorithm checks for bookmarks flagged as `removed`, then checks if their `deleted_at` field is older than 14 days (to be added). If that's the case the entry gets garbage collected. 

So to sum it up we need to:

1. Add a `deleted_at` field to `bookmarks`
2. Ensure that the field is updated when `RemoveBookmark` RPC APIs are called
3. Implement the garbage collection described above and have it run during `Messenger.Start()`
